### PR TITLE
Support non POST methods for Lifecycle and Admin APIs

### PR DESCRIPTION
**Pull request by bharaththiruveedula**
Signed-off-by: Bharath Thiruveedula <bharath_ves@hotmail.com>

### CODE REVIEW

**Comment by bharaththiruveedula** [comment]
Implements #5343

**Comment by brian-brazil** [comment]
This should apply to the other admin endpoints too.

**Comment by bharaththiruveedula** [comment]
@brian-brazil you mean  for /quit also?

**Comment by brian-brazil** [comment]
Both.

File: prompb/rpc.pb.gw.go [hunk 1]
```diff
@@ -45,6 +45,23 @@ func request_Admin_TSDBSnapshot_0(ctx context.Context, marshaler runtime.Marshal
 
 }
 
+var (
```
Comment (brian-brazil) [inline prompb/rpc.pb.gw.go]: There's no need to update all this, v1 is where it's at.

**Comment by bharaththiruveedula** [comment]
> The docs should also be updated.

No Need to update v2?

File: prompb/rpc.pb.gw.go (hunk 1 above)
Comment (bharaththiruveedula) [reply prompb/rpc.pb.gw.go]: No Need to update V2 apis?

Comment (brian-brazil) [reply prompb/rpc.pb.gw.go]: No need.

**Comment by bharaththiruveedula** [comment]
> The docs should also be updated.
Can you point me which docs to be updated. Can I update accordingly this doc https://github.com/prometheus/prometheus/blob/master/docs/configuration/configuration.md

**Comment by brian-brazil** [comment]
It's docs/querying/api.md

**Comment by bharaththiruveedula** [comment]
> It's docs/querying/api.md

Done

File: docs/querying/api.md [hunk 2]
```diff
@@ -733,6 +745,12 @@ Example:
 $ curl -X POST \
   -g 'http://localhost:9090/api/v1/admin/tsdb/delete_series?match[]=up&match[]=process_start_time_seconds{job="prometheus"}'
 ```
+
+```json
+$ curl -X PUT \
+  -g 'http://localhost:9090/api/v1/admin/tsdb/delete_series?match[]=up&match[]=process_start_time_seconds{job="prometheus"}'
+```
+
```
Comment (brian-brazil) [inline docs/querying/api.md]: You should add a note here that PUT was added in 2.9

File: docs/querying/api.md [hunk 3]
```diff
@@ -733,6 +745,12 @@ Example:
 $ curl -X POST \
   -g 'http://localhost:9090/api/v1/admin/tsdb/delete_series?match[]=up&match[]=process_start_time_seconds{job="prometheus"}'
 ```
+
+```json
+$ curl -X PUT \
```
Comment (brian-brazil) [inline docs/querying/api.md]: You can have just the POST example

File: docs/querying/api.md [hunk 4]
```diff
@@ -750,4 +769,45 @@ This takes no parameters or body.
 $ curl -XPOST http://localhost:9090/api/v1/admin/tsdb/clean_tombstones
 ```
 
+```json
+$ curl -XPUT http://localhost:9090/api/v1/admin/tsdb/clean_tombstones
+```
+
 *New in v2.1*
+
+## Prometheus Lifecycle APIs
```
Comment (brian-brazil) [inline docs/querying/api.md]: This page is scoped to /v1, which these aren't under. And apparently quit isn't documented anywhere. This seems like something to leave for another PR, as we'll need to find a space for these.

Comment (bharaththiruveedula) [reply docs/querying/api.md]: Okay. I will remove this section

File: docs/querying/api.md (hunk 3 above)
Comment (bharaththiruveedula) [reply docs/querying/api.md]: I will remove PUT example

File: docs/querying/api.md (hunk 2 above)
Comment (bharaththiruveedula) [reply docs/querying/api.md]: Done

File: docs/querying/api.md [hunk 5]
```diff
@@ -750,4 +752,4 @@ This takes no parameters or body.
 $ curl -XPOST http://localhost:9090/api/v1/admin/tsdb/clean_tombstones
 ```
 
-*New in v2.1*
+*New in v2.1*
```
Comment (brian-brazil) [inline docs/querying/api.md]: The notes are generally put here (presuming we're following Python), so it'd be PUT is new in v2.9 or similar.

Comment (bharaththiruveedula) [reply docs/querying/api.md]: Fixed

**Comment by brian-brazil** [comment]
Thanks!
